app/views/admin/versions/_object_desc_and_link.html.haml

```diff
@@ -1,156 +1,163 @@
-- if version.item_type == 'UsersRole'
-  - users_role = get_version_object(version)
+- case version.item_type
+- when 'UsersRole'
+  - users_role = current_or_last_object_state(version.item_type, version.item_id)
+  - user = current_or_last_object_state('User', users_role.user_id)
   = 'role'
   = link_to users_role.role.name, admin_conference_role_path(conference_id: Conference.find(version.conference_id).short_title, id: users_role.role.name)
   = version.event == 'create' ? 'to' : 'from'
   = 'user'
-  = link_to users_role.user.name, admin_user_path(id: users_role.user.id)
 
-- elsif version.item_type == 'Subscription' || version.item_type == 'Registration'
+  = link_to (user.try(:name) || 'deleted user'), admin_user_path(id: users_role.user.id)
+
+- when 'Subscription', 'Registration'
   = 'conference'
   = link_to Conference.find(version.conference_id).title,
           admin_conference_registrations_path(conference_id: Conference.find(version.conference_id).short_title)
 
-- elsif version.item_type == 'Commercial'
-  - commercial_last_version = get_version_object(PaperTrail::Version.where(item_type: version.item_type, item_id: version.item_id).last)
-  - commercialable_last_version = get_version_object(PaperTrail::Version.where(item_type: commercial_last_version.commercialable_type,
-                                                          item_id: commercial_last_version.commercialable_id).last)
+- when 'Commercial'
+  - commercial = current_or_last_object_state(version.item_type, version.item_id)
+  - commercialable = current_or_last_object_state(commercial.commercialable_type, commercial.commercialable_id)
 
-  - case commercial_last_version.commercialable_type
+  - case commercial.commercialable_type
   - when 'Event'
-    =  link_to 'commercial',
-            edit_admin_conference_program_event_path(conference_id: Conference.find(version.conference_id).short_title,
-                id: commercialable_last_version.id, anchor: 'commercials-content')
-    = "in event #{commercialable_last_version.title}"
+    commercial in event
+    =  link_to (commercialable.try(:title) || 'deleted event'),
```

Comment:
It doesn't make sense to link to an event that does not exist and is going to raise an error anyway :/

If there is an event, we can link to it, just like you do.
If there is no event, we should show just the text (and possible the ID of the event: `Deleted event with ID 123`)